### ISSUE TITLE
Subsequent `show` rules do not override previous ones

### ISSUE THREAD

**Issue by ndtodoroff** [typst/typst#351]
```typst
#show math.equation: set text(blue)
#show math.equation: set text(red)

$1 + 1 = 2$
```
This yields a blue equation, not a red equation. `set` rules do override properly:
```typst
#set text(blue)
#set text(red)

$1 + 1 = 2$
```
This yields a red equation.

It's some issue with show itself because the "expanded" form doesn't work either:
```typst
#show math.equation: it => {
  set text(blue)
  it
}
#show math.equation: it => {
  set text(red)
  it
}

$1 + 1 = 2$
```
This yields a blue equation.

**Comment by laurmaedje** [comment]
Yeah, that's a somewhat hard-to-fix bug, that I will try to tackle soon.

**Comment by Complex2-Liu** [comment]
Could someone please explain the **difference** between the "normal" `sho x: set y` rule and the "expanded" form

```typst
#show math.equation: it => {
  set text(blue)
  it
}
```

**Comment by laurmaedje** [comment]
It's pretty much equivalent, which is precisely the problem here.

**Comment by Phantop** [comment]
Is there any workaround to this besides having to scope things such that the first rule's scope ends before the second?

**Comment by Andrew15-5** [comment]
I don't think so, but a hacky way can **potentially** always exist (meaning that **sometime** you can't make your way around the limitations even if you try very hard). Everyone is waiting for the v0.11.0 which should go public somewhere in the February 2024. After it a lot of limitations should disappear or they can be fixed very easily (after v0.11.0).
